### ISSUE TITLE
[Feature] Server option to hide 'Map Explorer'

### ISSUE THREAD

**Issue by SterLovesFood** [RPTools/maptool#3224]
Feature Request
With the new additions in 1.11.0 in disabling the Library from connected clients I see no reason this can't also be applied to the 'Map Explorer' tab. The map explorer allows players to see what monster tokens are prepared on the map, trap tiles, etc and spoil themselves. I've had players do this by accident as well as the occasional person who can't help themselves once they know the Map Explorer exists.

The Solution you'd like
Ability to toggle the 'Map Explorer' window for connecting clients when creating a server, just as the 'Hide Library' feature that was introduced in 1.11.0

**Comment by FullBleed** [comment]
That doesn't sound right.  As far as I know, they shouldn't be able to see anything in the map explorer that they don't own.

So if they can, that doesn't sound like a feature request it sounds like a bug.  Will need some verification of this though...

That said, I know that at least one addon ("The Doors" @bubblobill) does set doors to being owned by all so players can access them... and GM's might want to do that for other objects too and having players use the Map Explorer to locate some of those things before on-map discovery certainly would not be intended... so, you're right, this needs to be a Feature Request, too.

@michaelperino did the Library Disabling feature so perhaps his arm can be twisted to do a Map Explorer disable, too.

**Comment by michaelperino** [comment]
Man the start server dialog is going to turn into a multi page endeavor at this rate 😅. Probably pretty easy to implement, however work is picking up before the holidays so my bandwidth is a little limited until mid December.

**Comment by FullBleed** [comment]
> Man the start server dialog is going to turn into a multi page endeavor at this rate 😅. Probably pretty easy to implement, however work is picking up before the holidays so my bandwidth is a little limited until mid December.

Heh... Yeah I think the start server dialog probably needs to be organized a little better, too.  I suggested something a while back about better grouping of options... but never did a mockup for it... I just did now though.  I don't have 1.11 installed to see the new lines, so I just added a "Permissions" section that could head the selections.

Of course, I don't know how much trouble adding a second column under options would be, but I think this is much better organized and helps mitigate Start Server growth by actually providing more efficient use of screen real-estate.

![Start Server WiP](https://user-images.githubusercontent.com/37131742/144201125-5e4b3aa3-53c6-4361-b6f7-7f4d0caad6f4.jpg)

**Comment by Phergus** [comment]
>  The map explorer allows players to see what monster tokens are prepared on the map, trap tiles, etc and spoil themselves. I've had players do this by accident as well as the occasional person who can't help themselves once they know the Map Explorer exists.

This would only happen if you don't turn on Strict Token Ownership which means not only can the Players see them but they can also edit them, delete them and move them from the map view.  Not sure why you would want your players to be able to do that.

**Comment by michaelperino** [comment]
> Of course, I don't know how much trouble adding a second column under options would be, but I think this is much better organized and helps mitigate Start Server growth by actually providing more efficient use of screen real-estate.

Is that just a Photoshop mockup? I don't think it's hard to put in a new column. It's even easier to just do a tabbed pane, so unless anyone has a huge problem with that it's probably the easiest to implement way.

**Comment by FullBleed** [comment]
> Is that just a Photoshop mockup? I don't think it's hard to put in a new column. It's even easier to just do a tabbed pane, so unless anyone has a huge problem with that it's probably the easiest to implement way.

Yes.  I tried to use realistic spacing (same current fixed width, option box width, and line spacing between current options.)

The problem with a tabbed pane is that we'd be getting into a situation there where users would have to look through tabs to get to all of the settings so I think keeping it all on one page is still the best option if possible (until we no choice but to go with tabs!)  The mockup I used was supposed to illustrate that it could fit in the same space as the server start screen in 1.10.4 while also being more organized (because as new options have been added over time there hasn't been much effort to group them in any purposeful way.)

**Comment by Phergus** [comment]
One thing to keep in mind about trying to do this with columns is that some of the languages take up considerably more space in the UI.  German in particular takes up more space but also Italian.  Others would if their translations were near complete.

As these settings are not often changed by users moving the various UI permissions to their own tab won't be particularly onerous.

**Comment by FullBleed** [comment]
I agree it's usually set once and then rarely changed after that.  But from the perspective of the first time server setup or newer users I do think having to go through all the tabs to setup a server might seem more complicated than it is... and it might also lead some to just going with the defaults.

On more complicated pieces of software I know I usually go through all the setup tabs... but I suspect many companies hope you don't (in particular when they "hide" their use/data collection opt-outs in them.)  That, I think, suggests that many users don't go tab surfing.  Don't really care one way another for myself... just see it potentially being something that leads to a lot of answers to questions that start with, "When setting up the server go to the XXX tab and..."

**Comment by melek** [comment]
I think tabs are a good idea. The dialog is busy and as we get more options I think that it serves users to organize it.

Rather break it into tabs than have a hard to understand wall of options, and if tabs lead to a more predictable localization situation I like them more than columns.

**Comment by thelsing** [comment]
I think it would be best to hide all those options per default and only expand them on a "Configure Server"-Button. I only configure the server once for my style of play and usually don't change it for later sessions.

**Comment by Phergus** [comment]
For the record also organizing them like FullBleed did in his mockup would also improve things significantly.

**Comment by FullBleed** [comment]
> I think it would be best to hide all those options per default and only expand them on a "Configure Server"-Button. I only configure the server once for my style of play and usually don't change it for later sessions.

If we're going with tabs... I'd be more on this page... with maybe one "Configure" tab.  So we'd have a default "Connection" tab and a "Configure" tab.

But I'd still probably would go with two columns since I think it's a more efficient use of space.  ;)

One last consideration for keeping it all on one page... seeing the options may be a little busy, but it is also a persistent reminder of what MT can do.  "Out of sight out of mind."
